jcl/src/java.base/share/classes/jdk/crac/Core.java

```diff
@@ -69,7 +74,12 @@ class CRIUSupportContext<R extends Resource> extends Context<R> {
 	// InternalCRIUSupport.getCRaCCheckpointToDir() is not null if
 	// InternalCRIUSupport.isCRaCSupportEnabled() returns true before creating CRIUSupportContext<>().
 	private final InternalCRIUSupport internalCRIUSupport = new InternalCRIUSupport(
-			Paths.get(InternalCRIUSupport.getCRaCCheckpointToDir())).setLeaveRunning(false).setShellJob(true)
+			Paths.get(InternalCRIUSupport
+			.getCRaCCheckpointToDir()))
+			.setLeaveRunning(false)
+			.setShellJob(true)
+			.setTCPEstablished(true)
+			.setUnprivileged(true)
```

Comment:
`tcpEstablished` and `unprivileged` are `false` by default, does `CRaC` application always expect `true`?

Reply:
its not a crac app requirement per se, its a usability requirement for criu itself. One cant realistically use criu with privileged access in production (Liberty doesnt either). Spring boot has modes that takes checkpoint late, which means that a tcp connection is likely already made. Thats why Ive change the default fro Crac APIs, but added options to override them if its not desired.